lib/matplotlib/widgets.py

```diff
@@ -621,6 +622,247 @@ def disconnect(self, cid):
+
+    For the text box to remain responsive
+    you must keep a reference to it.
+
+    The following attributes are accessible
+
+      *ax*
+        The :class:`matplotlib.axes.Axes` the button renders into.
+
+      *label*
+        A :class:`matplotlib.text.Text` instance.
+
+      *color*
+        The color of the text box when not hovering.
+
+      *hovercolor*
+        The color of the text box when hovering.
+
+    Call :meth:`on_text_change` to be updated whenever the text changes
+    Call :meth:`on_submit` to be updated whenever the user hits enter or leaves the text entry field
+    """
+
+    def __init__(self, ax, label, initial = '', 
+                 color='.95', hovercolor='1'):
+        """
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            The :class:`matplotlib.axes.Axes` instance the button
+            will be placed into.
+
+        label : str
+            Label for this text box. Accepts string.
+        
+        initial : str
+            Initial value in the text box
+            
+        color : color
+            The color of the box
+
+        hovercolor : color
+            The color of the box when the mouse is over it
+        """
+        AxesWidget.__init__(self, ax)
+        
+        self.DIST_FROM_LEFT = .05        
+        
+        self.params_to_disable = []
+        for key in rcParams.keys():                   
+            if u'keymap' in key:
+               self.params_to_disable += [key]
+        
+        self.text = initial
+        self.label = ax.text(0.0,0.5, label,
+                             verticalalignment='center',
+                             horizontalalignment='right',
+                             transform=ax.transAxes)
+        self.text_disp = self._make_text_disp(self.text)
+        
+        self.cnt = 0
+        self.change_observers = {}
+        self.submit_observers = {}
+        
+        self.ax.set_xlim(0, 1)      #If these lines are removed, the cursor won't appear 
+        self.ax.set_ylim(0, 1)      #the first time the box is clicked
+        
+        self.cursor_index = 0;
+        self.cursor = self.ax.vlines(0, 0, 0)    #because this is initialized, _render_cursor 
+        self.cursor.set_visible(False)           #can assume that cursor exists
+        
+        self.connect_event('button_press_event', self._click)
+        self.connect_event('button_release_event', self._release)
+        self.connect_event('motion_notify_event', self._motion)
+        self.connect_event('key_press_event', self._keypress)
+        ax.set_navigate(False)
+        ax.set_axis_bgcolor(color)
+        ax.set_xticks([])
+        ax.set_yticks([])
+        self.color = color
+        self.hovercolor = hovercolor
+
+        self._lastcolor = color
+        
+        self.capturekeystrokes = False        
+           
+    def _make_text_disp(self, string):
+        return self.ax.text(self.DIST_FROM_LEFT, 0.5, string,
```

Comment:
This spacing should be in display, not data coordinates.  That will be tricky and may require the use of the inverse transform, but will result in spacing that doesn't change with the size of the figure.